types/route.ts

```diff
@@ -179,6 +191,70 @@ export class Route extends LazyObject {
     return Promise.all(tasks);
   }
 
+  /** edit
+   * Update a route with any of the non-undefined params.
+   * All params are optional since not every param *has* to change.
+   * @param name: Route's name
+   * @param classifier: Route's classifier
+   * @param description: Optional, the route's description
+   * @param tags: Optional, a list of Tag, the route's tags
+   * @param setter: Optional, the Tower User of the setter
+   * @param rope: Optional, which rope the route is on / closest to
+   * @param thumbnail: Optional, the route's thumbnail
+   * @param color: Optional, the hold colors
+   * @param setterRawName: Optional, if no setter User exists, then just the name of the setter
+   * @remarks Updates this route's fields
+   */
+  public async edit({
+    name = undefined,
+    classifier = undefined,
+    description = undefined,
+    tags = undefined,
+    setter = undefined,
+    rope = undefined,
+    thumbnail = undefined,
+    color = undefined,
+    setterRawName = undefined,
+  }: EditRouteArgs) {
+    await this.getData(true);
+    if (thumbnail) {
+      if (this.thumbnail) await deleteObject(this.thumbnail.getStorageRef());
+      await uploadBytes(
+        ref(storage, 'routeThumbnails/' + this.docRef!.id),
+        thumbnail
+      );
+    }
```

Comment:
we do all or nothing operations for DB stuff, do we want that on storage operations as well?
Any realistic cases where there's a downside to deleting before we know our upload will work?

Reply:
see discussion below
![image](https://user-images.githubusercontent.com/73561858/214179874-6bb08bed-47d7-4960-8d32-3f1a8824d430.png)